airbyte-integrations/connectors/source-s3/source_s3/v4/legacy_config_transformer.py

```diff
@@ -55,6 +60,36 @@ def _create_globs(cls, path_pattern: str, path_prefix: str) -> List[str]:
     def _transform_seconds_to_micros(cls, datetime_str: str) -> str:
         try:
             parsed_datetime = datetime.strptime(datetime_str, SECONDS_FORMAT)
-            return datetime.strftime(parsed_datetime, MICROS_FORMAT)
+            return parsed_datetime.strftime(MICROS_FORMAT)
         except ValueError as e:
-            raise e
+            raise ValueError("Timestamp could not be parsed when transforming legacy connector config") from e
+
+    @classmethod
+    def _transform_file_format(cls, format_options: Union[CsvFormat, ParquetFormat, AvroFormat, JsonlFormat]) -> Mapping[str, Any]:
+        if isinstance(format_options, AvroFormat):
+            return {"filetype": "avro"}
+        elif isinstance(format_options, CsvFormat):
+            csv_options = {
+                "filetype": "csv",
+                "delimiter": format_options.delimiter,
+                "quote_char": format_options.quote_char,
+                "double_quote": format_options.double_quote,
+                "null_values": ["", "null", "NULL", "N/A", "NA", "NaN", "None"],
+                "true_values": ["y", "yes", "t", "true", "on", "1"],
+                "false_values": ["n", "no", "f", "false", "off", "0"],
+                "inference_type": "Primitive Types Only" if format_options.infer_datatypes else "None",
+                "strings_can_be_null": True,
```

Comment:
`strings_can_be_null` should be set from CsvSpec.additional_reader_options["strings_can_be_null"] or default to false https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html#pyarrow.csv.ConvertOptions

Reply:
wait so just to confirm, we want it default to false when its not in the additional options?

In the PR description we talked about during planning was: `strings_can_be_null -> default to True in legacy`

Reply:
yeah - I must have I derped when we groomed the issue.

- [default to true in the cdk](https://github.com/airbytehq/airbyte/pull/29298/files#diff-5452fd4d2184531c1eeaa4b6aee9743293e2b399754933dc3b0b7d6beb7ff092R69)
- default to false in legacy because [that's the default in pyarrow](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html#pyarrow.csv.ConvertOptions)